src/parse/Parser.zig

```diff
@@ -536,6 +537,30 @@ pub fn parseMaybeNativeFunctionDeclStatement(self: *Self) !?usize {
     });
 }
 
+pub fn parseMaybeReturnStatement(self: *Self) !?usize {
+    if (try self.maybe(.KW_RETURN) == null) return null; // This may not be a return statement.
+    try self.pushSpan();
+    defer _ = self.popSpan();
+
+    var expression: ?usize = null;
+
+    if ((try self.peek()).type != .SEMICOLON) {
+        // If we do not have semicolon, we expect an expression.
+        expression = try self.parseExpression();
+        _ = try self.expect(.SEMICOLON);
+    } else {
+        // If we have semicolon, we do not return anything.
+        _ = try self.expect(.SEMICOLON);
+    }
```

Comment:
You could use `.maybe(.SEMICOLON)` to avoid having an else branch:

```suggestion
    if (try self.maybe(.SEMICOLON) == null) {
        // If we do not have semicolon, we expect an expression.
        expression = try self.parseExpression();
        _ = try self.expect(.SEMICOLON);
    }
```